kentikapi/client.go

```diff
@@ -0,0 +1,75 @@
+package kentikapi
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/kentik/community_sdk_golang/kentikapi/models"
+)
+
+// Public constants.
+const (
+	APIURLUS = "https://api.kentik.com/api/v5"
+	APIURLEU = "https://api.kentik.eu/api/v5"
+)
+
+// Client is Kentik API client.
+type Client struct {
+	config     ClientConfig
+	restClient *restClient
+}
+
+// ClientConfig holds configuration of the client.
+type ClientConfig struct {
+	// APIURL defaults to "https://api.kentik.com/api/v5"
+	APIURL    string
+	AuthEmail string
+	AuthToken string
+}
+
+// NewClient creates a new client.
+func NewClient(c ClientConfig) *Client {
+	if c.APIURL == "" {
+		c.APIURL = APIURLUS
+	}
+
+	return &Client{
+		config: c,
+		restClient: newRestClient(restClientConfig{
+			APIURL:    c.APIURL,
+			AuthEmail: c.AuthEmail,
+			AuthToken: c.AuthToken,
+		}),
+	}
+}
+
+// GetAllUsers lists users.
+func (c *Client) GetAllUsers(ctx context.Context) (_ []models.User, err error) {
+	responseBody, err := c.restClient.Get(ctx, "/users")
```

Comment:
Done: https://github.com/kentik/community_sdk_golang/pull/1#discussion_r574421291

Reply:
Mixed levels of abstraction: do you mean that HTTP path is too low level (transport) to fit here?

I extracted constants and a function. Take a look if it is satisfactory

Reply:
I mean gather the URL stings constructors in separate file/package. There will be ~50 of them scattered across all the api classes otherwise. eg file: apiendpoints.go, or we can stick to separation by resource type and store endpoints in groups per/resource eg. under apiendponts/users.go

Reply:
Moved to `api_endpoints.go`